test/Staking.test.ts

```diff
@@ -357,37 +365,99 @@ describe('Staking', function () {
+    it('should not allow stake migration while unpaused', async function () {
       await mintAndApprove(staker_0, stakeRegistry.address, stakeAmount_0);
       await sr_staker_0.manageStake(nonce_0, stakeAmount_0);
-
-      await expect(sr_staker_0.withdrawFromStake(stakeAmount_0)).to.be.revertedWith(errors.pause.notCurrentlyPaused);
+      await expect(sr_staker_0.migrateStake()).to.be.revertedWith(errors.pause.notCurrentlyPaused);
     });
 
-    it('should allow stake withdrawal while paused', async function () {
+    it('should allow stake migration while paused', async function () {
       const staked_before = await sr_staker_0.stakes(staker_0);
 
       expect(staked_before.overlay).to.be.eq(overlay_0);
-      expect(staked_before.stakeAmount).to.be.eq(stakeAmount_0);
+      expect(staked_before.potentialStake).to.be.eq(stakeAmount_0);
       expect(staked_before.lastUpdatedBlockNumber).to.be.eq(updatedBlockNumber);
 
       const stakeRegistryPauser = await ethers.getContract('StakeRegistry', pauser);
       await stakeRegistryPauser.pause();
 
       expect(await token.balanceOf(staker_0)).to.be.eq(zeroAmount);
 
-      await sr_staker_0.withdrawFromStake(stakeAmount_0);
+      await sr_staker_0.migrateStake();
 
       const staked_after = await sr_staker_0.stakes(staker_0);
 
       expect(await token.balanceOf(staker_0)).to.be.eq(stakeAmount_0);
 
       expect(staked_after.overlay).to.be.eq(zeroBytes32);
-      expect(staked_after.stakeAmount).to.be.eq(zeroStake);
+      expect(staked_after.potentialStake).to.be.eq(zeroStake);
       expect(staked_after.lastUpdatedBlockNumber).to.be.eq(0);
 
       await stakeRegistryPauser.unPause();
     });
+
+    it('should not allow deposit stake below minimum', async function () {
+      const sr_staker_1 = await ethers.getContract('StakeRegistry', staker_1);
+      await mintAndApprove(staker_1, stakeRegistry.address, stakeAmount_1_n);
+      await expect(sr_staker_1.manageStake(nonce_1, stakeAmount_1_n)).to.be.revertedWith(errors.deposit.belowMinimum);
+    });
+
+    it('should make stake surplus withdrawal', async function () {
+      const staked_before = await sr_staker_0.stakes(staker_0);
+      const priceOracle = await ethers.getContract('PriceOracle', deployer);
+
+      expect(staked_before.overlay).to.be.eq(overlay_0);
+      expect(staked_before.potentialStake).to.be.eq(stakeAmount_0);
+      expect(staked_before.lastUpdatedBlockNumber).to.be.eq(updatedBlockNumber);
+
+      // Check that balance of wallet is 0 in the begining and lower the price
+      expect(await token.balanceOf(staker_0)).to.be.eq(zeroAmount);
+      await priceOracle.setPrice(24000);
+
+      await sr_staker_0.withdrawFromStake();
+      const staked_after = await sr_staker_0.stakes(staker_0);
+
+      expect(staked_before.potentialStake.gt(staked_after.potentialStake)).to.be.true;
+      expect(staked_after.potentialStake.toString()).to.be.eq('75000000000000000');
```

Comment:
the difference should be calculated here to check the correctness of the withdraw function and effective stake calculation in the contract.